src/components/Questions/QuestionModal.vue

```diff
@@ -292,7 +250,7 @@ To attempt Q.${props.currentQuestionIndex + 1}, unselect an answer to another qu
     function questionOptionSelected(answer: number | string) {
       if (isQuestionTypeSingleChoice.value && typeof answer == "number") {
         // for MCQ, simply set the option as the current response
-        state.draftResponses[props.currentQuestionIndex] = [answer]
+        state.draftResponses[props.randomIndex[props.currentQuestionIndex]] = [answer]
```

Comment:
instead of doing props.randomIndex[props.currentQuestionIndex] everywhere.. probably store this value in some computed variable.. currentShuffledIndex maybe